### PR TITLE
Book comments 👽

### PR DESCRIPTION
**Pull request by moralesbang**
### Summary

Added comments in book-detail.

### Screenshots

Screen Shot 2020-05-26 at 12 37 54 PM

### Trello Card

https://trello.com/c/HijFXo1Z

### CODE REVIEW

File: src/app/components/CommentCard/index.tsx [hunk 1]
```diff
@@ -0,0 +1,21 @@
+import React from 'react';
+import { View, Image, Text } from 'react-native';
+import { Comment } from '@interfaces/comment';
+
+import styles from './styles';
+
+interface Props extends Comment {}
+
+function CommentCard({ avatar, author, content }: Props) {
```
Comment (sfernandez11) [inline src/app/components/CommentCard/index.tsx:9]: Move this component inside `CommentList/components` or are you going to use it from another screen? Same with `BookCard` => `app/screens/BookList/components/BookCard`. Both components will only be used in one screen.

File: src/app/components/CommentCard/index.tsx [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+  return (
+    <View style={styles.comment}>
+      <Image source={avatar} style={styles.avatar} />
```
Comment (sfernandez11) [inline src/app/components/CommentCard/index.tsx:12]: Missing `resizeMode` for the Image

File: src/app/components/CommentCard/styles.ts [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+    marginBottom: 8
+  },
+  commentContent: {
+    color: gray,
+    fontSize: 16,
+    lineHeight: 20
+  },
+  avatar: {
+    width: 45,
+    height: 45,
```
Comment (sfernandez11) [inline src/app/components/CommentCard/styles.ts:26]: Extract the common value:
```
const AVATAR_SIZE = 45;
...
    width: AVATAR_SIZE,
    height: AVATAR_SIZE,
...
```

File: src/app/components/CommentCard/styles.ts [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+    borderRadius: 45 / 2,
```
Comment (sfernandez11) [inline src/app/components/CommentCard/styles.ts:27]: `borderRadius: AVATAR_SIZE / 2,`

File: src/app/screens/BookDetail/components/CommentList/index.tsx [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+import styles from './styles';
+
+function CommentList() {
+  const renderComment = (comment: Comment) => <CommentCard key={comment.id} {...comment} />;
+  return (
+    <View style={styles.commentSection}>
+      <View>{COMMENTS.map(renderComment)}</View>
+      <TouchableOpacity style={styles.viewAllButton}>
+        <Text style={styles.viewAllButtonText}>View All</Text>
+      </TouchableOpacity>
```
Comment (sfernandez11) [inline src/app/screens/BookDetail/components/CommentList/index.tsx:16]: What happens when the user taps this button?

File: src/app/screens/BookDetail/components/CommentList/styles.ts [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+import { white, blue } from '@constants/colors';
+
+export default StyleSheet.create({
+  commentSection: {
+    backgroundColor: white,
+    paddingHorizontal: 16,
+    paddingVertical: 20,
+    borderRadius: 5
+  },
+  commentList: {},
```
Comment (sfernandez11) [inline src/app/screens/BookDetail/components/CommentList/styles.ts:11]: ???

File: src/app/screens/BookDetail/index.tsx [hunk 7]
```diff
@@ -14,30 +15,33 @@ interface Props {
...
+          </View>
+          <View style={styles.bookDetailActions}>
+            <TouchableOpacity style={styles.watchlistButton}>
+              <Text style={styles.watchlistButtonText}>ADD TO WATCHLIST</Text>
+            </TouchableOpacity>
+            <TouchableOpacity style={styles.rentButton}>
+              <Text style={styles.rentButtonText}>RENT</Text>
+            </TouchableOpacity>
           </View>
         </View>
```
Comment (sfernandez11) [inline src/app/screens/BookDetail/index.tsx:41]: Extract these lines into a separate Component

File: src/app/screens/BookDetail/components/CommentList/index.tsx (hunk 5 above)
Comment (moralesbang) [reply src/app/screens/BookDetail/components/CommentList/index.tsx:16]: Now, I undestand the point

File: src/app/components/CommentCard/index.tsx (hunk 2 above)
Comment (moralesbang) [reply src/app/components/CommentCard/index.tsx:12]: Default value for `resizeMode` is `cover`. That's is what I'm looking for.

File: src/app/screens/BookDetail/components/BookInfo/styles.ts [hunk 8]
```diff
@@ -0,0 +1,72 @@
...
+  },
+  bookStatus: {
+    color: red,
+    fontWeight: 'bold',
+    fontSize: 16,
+    lineHeight: 20
+  },
+  bookBasicAttribute: {
+    fontSize: 16,
+    lineHeight: 20,
```
Comment (sfernandez11) [inline src/app/screens/BookDetail/components/BookInfo/styles.ts:41]: For the future, you can extract the common props:
```
const MEDIUM_FONT = { fontSize: 16, lineHeight: 20 };
...
bookStatus: { ...MEDIUM_FONT, ... },
bookBasicAttribute: { ...MEDIUM_FONT, ... },
...
```
Just for comment, no need to change it now.

File: src/app/screens/BookDetail/components/CommentList/index.tsx [hunk 9]
```diff
@@ -0,0 +1,24 @@
...
+function CommentList() {
+  const [listIsExpanded, setlistIsExpanded] = useState(false);
+  const togglelistIsExpanded = () => setlistIsExpanded(!listIsExpanded);
+  const commentsLength = listIsExpanded ? 5 : 2;
+  const renderComment = (comment: Comment) => <CommentCard key={comment.id} {...comment} />;
+  return (
+    <View style={styles.commentSection}>
+      <View>{COMMENTS.slice(0, commentsLength).map(renderComment)}</View>
+      <TouchableOpacity onPress={togglelistIsExpanded} style={styles.viewAllButton}>
+        <Text style={styles.viewAllButtonText}>{listIsExpanded ? 'Show Less' : 'Show More'}</Text>
```
Comment (sfernandez11) [inline src/app/screens/BookDetail/components/CommentList/index.tsx:18]: 👍

File: src/app/screens/BookList/index.tsx [hunk 10]
```diff
@@ -1,6 +1,6 @@
 import React from 'react';
 import { FlatList, ListRenderItem } from 'react-native';
-import BookCard from '@components/BookCard';
+import BookCard from '@app/screens/BookList/components/BookCard';
```
Comment (sfernandez11) [inline src/app/screens/BookList/index.tsx:3]: `import BookCard from './components/BookCard';`

File: src/constants/styleHelpers.ts [hunk 11]
```diff
@@ -0,0 +1,12 @@
+import { black } from './colors';
+
+export const SHADOW = {
```
Comment (sfernandez11) [inline src/constants/styleHelpers.ts:3]: 👏

File: src/app/screens/BookDetail/components/BookInfo/styles.ts (hunk 8 above)
Comment (moralesbang) [reply src/app/screens/BookDetail/components/BookInfo/styles.ts:41]: Thanks so much!

**Pull request merged**